File: frontend/src/types/report.ts
import { Scan } from "./scan";

export interface Report {
    id: string;
    scan_id: string;
    scan: Scan;
    ai_analysis: AIAnalysis | null;
    pdf_path: string | null;
    created_at: string;
    overall_score?: number;
    module_scores?: Record<string, number>;
    issues?: Issue[];
}

export interface AIAnalysis {
    executive_summary: string;
    overall_score: number;
    scores_by_category: Record<string, number>;
    critical_issues: Issue[];
    warnings: Issue[];
    recommendations: Recommendation[];
    performance_analysis: string;
    security_analysis: string;
    seo_analysis: string;
}

export interface Issue {
    title: string;
    description: string;
    category: string;
    severity: string;  // 'critical', 'high', 'medium', 'low', 'info'
    solution: string;
}

export interface Recommendation {
    priority: 'high' | 'medium' | 'low';
    title: string;
    description: string;
    impact: string;
}

File: backend/app/schemas/report.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel


# ──────────────────────────────────────────────────────
# Report Schemas
# ──────────────────────────────────────────────────────

class ReportCreate(BaseModel):
    scan_id: uuid.UUID
    ai_analysis: dict[str, Any] | None = None
    pdf_path: str | None = None


class ReportSchema(ReportCreate):
    id: uuid.UUID
    email_sent: bool
    email_sent_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class AIScanAnalysis(BaseModel):
    """Structured shape of the AI-generated analysis stored in JSONB."""
    executive_summary: str
    overall_score: int
    scores_by_category: dict[str, int]
    critical_issues: list[dict[str, Any]]
    warnings: list[dict[str, Any]]
    passed_checks: list[dict[str, Any]]
    recommendations: list[dict[str, Any]]
    performance_analysis: str
    security_analysis: str
    seo_analysis: str
